### PR TITLE
Added new plugins files via upload

### PR DESCRIPTION
**Pull request by jsmonzani**
New plugins: bipsi color, import lvllvl and import Tiled, with sample files

### CODE REVIEW

File: src/plugin-import-lvllvl.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+1.2 Either design your rooms as a single long vertical image, or as frames in the animation timeline (bottom of the screen). For instance, a room is by default 16x16, and you could create a 16x32 image for two rooms or a 16x16 image with two animation frames
+1.3 Export your tileset to PNG (Tiles > Save Tile Set)
+1.4 Export your JSON as an image (Export > Export JSON and use the Native format. Tile Set Data isn't used so you can uncheck this)
+2. Import plugin in bipsi
+3. Go to edit mode, you'll find new buttons on the bottom right. First import your tilset, then the JSON image
+
+*/
+
+//! CODE_EDITOR
+//! CODE_EDITOR
```
Comment (seleb) [inline src/plugin-import-lvllvl.js:26]:
```suggestion
//! CODE_EDITOR
```

File: src/plugin-import-lvllvl.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                    y = 0;
+                    room++;
+                }	
+            }
+        }         
+        EDITOR.requestRedraw();
+    });
+}
+
+//! CODE_EDITOR
```
Comment (seleb) [inline src/plugin-import-lvllvl.js:98]:
```suggestion
```

File: src/plugin-import-lvllvl.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        }
+    }
+    return await loadImage(output.canvas.toDataURL());
+}
+
+var imageUp = "<svg width=\"22\" height=\"15\" fill=\"currentColor\" viewBox=\"0 0 22 15\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M9 4.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z\"/><path d=\"M5 0a2 2 0 0 0-2 2v10a2 2 0 0 0 2 2h9.777A4.5 4.5 0 0 1 14 11.5a4.5 4.5 0 0 1 2.256-3.898l-2.033-1.05a.5.5 0 0 0-.577.094l-3.71 3.71-2.66-1.772a.5.5 0 0 0-.63.062L4 11V2a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v5.043A4.5 4.5 0 0 1 18.5 7a4.5 4.5 0 0 1 .5.03V2a2 2 0 0 0-2-2z\"/><path d=\"M18.495 15a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7zm.354-5.854 1.5 1.5a.5.5 0 0 1-.708.708l-.646-.647V13.5a.5.5 0 0 1-1 0v-2.793l-.646.647a.5.5 0 0 1-.708-.708l1.5-1.5a.5.5 0 0 1 .708 0z\"/></svg>";
+
+var caret = "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"16\" height=\"16\" fill=\"currentColor\" class=\"bi bi-caret-up-fill\" viewBox=\"0 0 16 16\"><path d=\"m7.247 4.86-4.796 5.481c-.566.647-.106 1.659.753 1.659h9.592a1 1 0 0 0 .753-1.659l-4.796-5.48a1 1 0 0 0-1.506 0z\"/></svg>";
+
+//! CODE_EDITOR
```
Comment (seleb) [inline src/plugin-import-lvllvl.js:164]:
```suggestion
```

File: src/plugin-import-lvllvl.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    toggle?.addEventListener("change", () => {
+        windowEl.hidden = !toggle.checked;
+    });
+    autoCloseToggledWindow(windowEl, toggle);
+    return {
+        window: windowEl,
+        button: toggleButtonEl,
+        toggle,
+    };
+}
```
Comment (seleb) [inline src/plugin-import-lvllvl.js:265]:
```suggestion
}

//! CODE_PLAYBACK
```

File: src/plugin-import-Tiled.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+2.1a Import/create your PNG tileset
+2.1b Optionally add animations to tiles. If so, export the tileset as JSON so that we can transfer the animation information to bipsi. Durations will be ignored.
+2.2a Design your map on a single layer. Make sure your map size matches bipsi's default 16x16. If you multiple rooms, create one layer for each.
+2.2b Export your map as JSON
+2. Import plugin in bipsi
+3. Go to edit mode, you'll find new buttons on the bottom right. First import your PNG tileset, then optionally your tiles animation (JSON), and the JSON map.
+*/
+
+//! CODE_EDITOR
+//! CODE_EDITOR
```
Comment (seleb) [inline src/plugin-import-Tiled.js:26]:
```suggestion
//! CODE_EDITOR
```

File: src/plugin-import-Tiled.js [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        for (tile of jsondata.tiles) {
+            data.tiles[tile.id].frames = tile.animation.map((x) => x.tileid); // ignore the duration attribute and keep the tileid
+        }
+
+    });
+}
+
+
+
+//! CODE_EDITOR
```
Comment (seleb) [inline src/plugin-import-Tiled.js:96]:
```suggestion
```

File: src/plugin-import-Tiled.js [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        }
+    }
+    return await loadImage(output.canvas.toDataURL());
+}
+
+var imageUp = "<svg width=\"22\" height=\"15\" fill=\"currentColor\" viewBox=\"0 0 22 15\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M9 4.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z\"/><path d=\"M5 0a2 2 0 0 0-2 2v10a2 2 0 0 0 2 2h9.777A4.5 4.5 0 0 1 14 11.5a4.5 4.5 0 0 1 2.256-3.898l-2.033-1.05a.5.5 0 0 0-.577.094l-3.71 3.71-2.66-1.772a.5.5 0 0 0-.63.062L4 11V2a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v5.043A4.5 4.5 0 0 1 18.5 7a4.5 4.5 0 0 1 .5.03V2a2 2 0 0 0-2-2z\"/><path d=\"M18.495 15a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7zm.354-5.854 1.5 1.5a.5.5 0 0 1-.708.708l-.646-.647V13.5a.5.5 0 0 1-1 0v-2.793l-.646.647a.5.5 0 0 1-.708-.708l1.5-1.5a.5.5 0 0 1 .708 0z\"/></svg>";
+
+var caret = "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"16\" height=\"16\" fill=\"currentColor\" class=\"bi bi-caret-up-fill\" viewBox=\"0 0 16 16\"><path d=\"m7.247 4.86-4.796 5.481c-.566.647-.106 1.659.753 1.659h9.592a1 1 0 0 0 .753-1.659l-4.796-5.48a1 1 0 0 0-1.506 0z\"/></svg>";
+
+//! CODE_EDITOR
```
Comment (seleb) [inline src/plugin-import-Tiled.js:162]:
```suggestion
```

File: src/plugin-import-Tiled.js [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+    toggle?.addEventListener("change", () => {
+        windowEl.hidden = !toggle.checked;
+    });
+    autoCloseToggledWindow(windowEl, toggle);
+    return {
+        window: windowEl,
+        button: toggleButtonEl,
+        toggle,
+    };
+}
```
Comment (seleb) [inline src/plugin-import-Tiled.js:280]:
```suggestion
}

//! CODE_PLAYBACK
```

File: src/plugin-import-lvllvl.js [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+    const [file] = await maker.pickFiles("application/json");
+    const mapdata = await maker.textFromFile(file);    
+    if (!file)
+        return;
+    // Create bipsi rooms & tiles
+    EDITOR.stateManager.makeChange(async (data) => {
+        const jsondata = JSON.parse(mapdata);
+        //import colors
+        let i = 1; //color 0 is reserved (transparent)
+        for (color of jsondata["colorPalette"]["data"]) {
```
Comment (seleb) [inline src/plugin-import-lvllvl.js:59]: when i tested this plugin with the provided sample, it broke on this line bc `jsondata` had no `colorPalette`, i'm not familiar with the format but i'm guessing it needs a nullish check before it tries to loop through here?

File: src/plugin-import-Tiled.js [hunk 10]
```diff
@@ -0,0 +1,280 @@
+/**
+🧱
```
Comment (seleb) [inline src/plugin-import-Tiled.js:2]: might be better to use the same emoji for all three since they're related? up to you though!

**Review by seleb — CHANGES_REQUESTED** [review]
a few things to fix:
- there are some errors with the `//! CODE_*` headers on the import plugins, see comments for fix
- there seems to be an error with the lvllvl import, see comment
- there are many lint errors
  - most of these are just formatting and can be resolved by running `npm run lint -- --fix`
  - most of the rest are "not defined" errors for things that need to be added to `globals` in `.eslintrc`
- inconsistent naming: i'd recommend a few renames to make the files more obviously related at a glance:
  - `plugin-import-Tiled.js` -> `bispi-color-import-Tiled.js`
  - `plugin-import-lvllvl.js` -> `bispi-color-import-lvllvl.js`
  - `sample-color-tilesets-and-Tiled-levels/*` -> `bispi-color-import-sample/*`

note that there's also quite a bit of shared code between the two import plugins. if you'd like, you can use es6 imports/exports to pull that out and reduce code duplication (there's no example of this in this repo atm, but you can see how it works in bitsy hacks which uses the same process: [export](https://github.com/seleb/bitsy-hacks/blob/f7032994abb73a4bb2e8cbbf31f30e6db24bfe4f/src/helpers/transform-sprite-data.js#L23) [import](https://github.com/seleb/bitsy-hacks/blob/f7032994abb73a4bb2e8cbbf31f30e6db24bfe4f/src/directional%20avatar.js#L18))

File: src/plugin-import-lvllvl.js (hunk 9 above)
Comment (jsmonzani) [reply src/plugin-import-lvllvl.js:59]: Yeah that's because I didn't provide a json file in lvllvl format, just one for Tiled.

**Comment by seleb** [comment]
closing as it was decided to merge these updates to a separate git

**Pull request closed without merging**